rfcs/001 - EC2 schema.md

```diff
@@ -0,0 +1,136 @@
+# 001 - EC2 Schema
+
+## Current Status
+
+### Proposed
+
+2022-07-18
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- LuisFer De Pombo <luisfer@iasql.com>
+- David Ellis <david@alantechnologies.com>
+- Yolanda Robla <yolanda@iasql.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Alejandro Guillen <alejandro@iasql.com>
+
+## Summary
+
+Adding new features to get closer to the goal of the EC2 module's completeness adds more complexity to the model. The `instance` table is the core of the `aws_ec2` module and is the one that can get affected the most. This complexity has a direct impact on the user experience, especially for new users, since they can feel overwhelmed if they encounter a complicated model.
+
+The definition of a solid base now will imply that new features added to this module will not translate into breaking changes to the model.
+
+Taking a look at the AWS EC2 console options, we would still need to add the following to reach EC2 module completeness:
+
+![AWS Console EC2](./assets/001/aws_console_ec2.png)
+
+- Instance types with Availability zone relationship
+- Launch templates
+- Spot instances
+- Dedicated hosts
+- Saving plans
+- Reserved instances
+- Scheduled instances
+- Capacity reservations
+- Auto Scaling Groups
+- AMIs management
+- EBS volume snapshots and life cycle
+- Placement groups
+- Key Pairs
+- Network interfaces
+
+The `instance` table still needs more advanced options to be defined. Some of those missing properties are: instance auto-recovery, shutdown behaviour, stop - hibernate behaviour, termination protection, stop protection, detailed cloudWatch monitoring, elastic GPU, credit specification, tenancy, RAM disk ID, kernel ID or Metadata accessible (I might be missing others). Additionally, some of the features listed above will need to be added to the `instance` table as FKs.
+
+The current EC2 model looks like this:
+
+![Current EC2 model](./assets/001/current_schema.png)
+Note: Could be also accessed via https://dbdocs.io/alejandro/iasql
```

Comment:
I'm curious, is this different from https://dbdocs.io/iasql/iasql?table=instance&schema=public&view=table_structure ?

Reply:
https://github.com/iasql/iasql-engine/issues/963 is prioritized this week which we can also use here for a `instance_simplified` type of table to provide a simpler UX that still allows peeking down the abstraction layer easily if/when the user needs to

Reply:
> I'm curious, is this different from https://dbdocs.io/iasql/iasql?table=instance&schema=public&view=table_structure ?

It is basically the same. I just picked that model and did some trim leaving just ec2 and tables directly related to it